src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -2486,7 +2486,7 @@ skip:
       drop_connection(context, true, false);
       return 1;
     }
-    if (arg.total_height < arg.m_block_ids.size() || arg.start_height > arg.total_height - arg.m_block_ids.size())
+    if (arg.total_height < arg.m_block_ids.size() || arg.start_height > arg.total_height - arg.m_block_ids.size() || arg.start_height >= m_core.get_current_blockchain_height())
```

Comment:
Just to clarify. Is it a response to our request, so starting height must always be below current blockchain height?

Reply:
It is.